### PR TITLE
config provider Service\DefaultNavigationFactory namespace fix

### PR DESCRIPTION
**Pull request by vaclavvanik**
Fix for [issue #31](https://github.com/zendframework/zend-navigation/issues/31).
Unit test added.

### CODE REVIEW

**Comment by RalfEggert** [comment]
Duplicate of https://github.com/zendframework/zend-navigation/pull/32

Since it adds an unit test should be prefered I guess

File: test/ConfigProviderTest.php [hunk 1]
```diff
@@ -0,0 +1,54 @@
+<?php
+/**
+ * Zend Framework (http://framework.zend.com/)
+ *
+ * @link      http://github.com/zendframework/zf2 for the canonical source repository
```
Comment (samsonasik) [inline test/ConfigProviderTest.php]: http://github.com/zendframework/zendframework

Comment (vaclavvanik) [inline test/ConfigProviderTest.php]: @samsonasik updated

File: test/ConfigProviderTest.php [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        $provider = new ConfigProvider();
+
+        $this->assertEquals($this->config, $provider->getDependencyConfig());
+    }
+
+    public function testInvoke()
+    {
+        $provider = new ConfigProvider();
+
+        $this->assertEquals(['dependencies' => $this->config], $provider());
```
Comment (weierophinney) [inline test/ConfigProviderTest.php]: I'd argue this should be:

```php
$this->assertEquals(['dependencies' => $provider->getDependencyConfig()], $provider());
```

Additionally, this test should depend on the `testGetDependencyConfig()` test as a result, as if that fails, there's no reason to run it.

I can make these changes on merge.

**Pull request merged**